pkg/agent/controller/service_export_client.go

```diff
@@ -104,6 +112,55 @@ func (c *ServiceExportClient) tryUpdateStatusConditions(ctx context.Context, nam
 	})
 }
 
+func (c *ServiceExportClient) mergeConflictCondition(to, from *mcsv1a1.ServiceExportCondition) bool {
+	var reasons, messages []string
+
+	if ptr.Deref(to.Reason, "") != "" {
+		reasons = strings.Split(ptr.Deref(to.Reason, ""), ",")
+	}
+
+	if ptr.Deref(to.Message, "") != "" {
+		messages = strings.Split(ptr.Deref(to.Message, ""), "\n")
+	}
+
+	index := goslices.Index(reasons, *from.Reason)
+	if index >= 0 {
+		if from.Status == corev1.ConditionTrue {
+			if index < len(messages) {
```

Comment:
When will this be true? Message is already present but not at end? Means this code is updating the message for this reason in case it changed?

Reply:
`index` should always be < `len(messages)` b/c it's size mirrors `reason`. This is just a sanity check.

If you're referring to `if from.Status == corev1.ConditionTrue`, yes, if the `reason` already exists but the `message` changed.